File: Kattis/wheresmyinternet.py
n, m = map(int, input().split())

l = [[] for _ in range(n)]

for _ in range(m):
    a, b = map(int, input().split())
    l[a-1].append(b-1)
    l[b-1].append(a-1)

visited = [False]*n
queue = [i for i in l[0]]
visited[0] = True

# print(queue, visited)

while queue:
    # print(visited, queue)
    i = queue.pop(0)
    if visited[i]:
        continue
    visited[i] = True
    for j in l[i]:
        if not visited[j]:
            queue.append(j)

r = [i+1 for i in range(n) if not visited[i]]

if r:
    print(*r,sep="\n")
else:
    print("Connected")

File: Kattis/wheresmyinternet/12247972/wheresmyinternet.cpp
#include <iostream>
#include <queue>
#include <vector>

int main(void)
{
    int n = 0, m = 0;
    std::cin >> n >> m;
    std::vector<std::vector<int> > graph(n);
    int a = 0, b = 0;
    for (int i = 0; i < m; i++) {
        std::cin >> a >> b;
        graph[a-1].push_back(b-1);
        graph[b-1].push_back(a-1);
    }
    std::vector<bool> visited(n, false);
    visited[0] = true;
    std::queue <int> q;
    for (int i = 0; i < graph[0].size(); i++)
        q.push(graph[0][i]);
    int i = 0;
    while (q.size() > 0) {
        i = q.front();
        q.pop();
        if (visited[i])
            continue;
        visited[i] = true;
        for (int j = 0; j < graph[i].size(); j++) {
            if (!visited[graph[i][j]])
                q.push(graph[i][j]);
        }
    }
    
    bool r = true;
    for (int j = 0; j < n; j++) {
        if (!visited[j]) {
            r = false;
            break;
        }
    }
    if (!r) {
        for (int j = 0; j < n; j++) {
            if (!visited[j]) {
                std::cout << j+1 << std::endl;
            }
        }
    } else {
        std::cout << "Connected" << std::endl;
    }
}